google/cloud/aiplatform/experimental/vertex_model/utils/source_utils.py

```diff
@@ -57,6 +64,38 @@ def _make_source(cls_source: str, cls_name: str, instance_method: str):
         between the user-written code and the string returned by this method is that
         the user has the option to specify a method to call from __main__.
     """
-    src = "\n".join(["import torch", "import pandas as pd", cls_source])
-    src = src + "if __name__ == '__main__':\n" + f"\t{cls_name}().{instance_method}()"
+    src = "\n".join(
+        [
+            "import torch",
+            "import pandas as pd",
+            "from google.cloud.aiplatform import training_util",
+            "from google.cloud.aiplatform.experimental.vertex_model.serializers import *",
+            cls_source,
+        ]
+    )
+
+    # First, add __main__ header
+    src = src + "if __name__ == '__main__':\n"
+
+    # Then, instantiate model
+    src = src + f"\tmodel = {cls_name}()\n"
+
+    src = src + f"\tmodel.{instance_method}("
+
+    # Iterate through parameters:
+    for (
+        parameter_name,
+        (parameter_uri, parameter_type),
+    ) in param_name_to_serialized_info.items():
+        deserializer = obj._data_serialization_mapping[parameter_type][0]
```

Comment:
I would add a comment here for future reference that we are currently working around not including the _serialization_mapping with our generated source and assume the serializer/deserializer is in a our serializer module.